lib/AST/ASTWalker.cpp

```diff
@@ -191,6 +191,20 @@ class Traversal : public ASTVisitor<Traversal, Expr*, Stmt*,
   }
 
   bool visitTypeAliasDecl(TypeAliasDecl *TAD) {
+    if (TAD->getGenericParams() &&
+        Walker.shouldWalkIntoGenericParams()) {
+      // Visit generic params
+      for (auto GP : TAD->getGenericParams()->getParams()) {
+        if (doIt(GP))
+          return true;
+      }
+      // Visit param conformance
+      for (auto &Req : TAD->getGenericParams()->getRequirements()) {
+        if (doIt(Req))
+          return true;
+      }
```

Comment:
Can we unify these two loops to ensure visit order of `Param1->Req1..->Param2->Req2...`?

Reply:
I'm not sure how I'd do that. I basically copied and pasted this from `visitNominalTypeDecl`

Reply:
Can we avoid copy-paste by extracting this as a function?

Reply:
Done in beb40e60be28069dd9b723f68842a6a66f6fb9fb.